hubverse_annotator/utils.py

```diff
@@ -264,68 +264,104 @@ def quantile_forecast_chart(
-    )
-    base_x_enc = alt.X(
-        "target_end_date:T",
-        axis=alt.Axis(title="Date", grid=grid, ticks=True, labels=True),
-    )
-    base_y_enc = alt.Y("median:Q", axis=y_axis, scale=alt.Scale(type=scale))
-    base = alt.Chart(df_wide, width=PLOT_WIDTH).encode(
-        x=base_x_enc,
-        y=base_y_enc,
-    )
-    band_95 = base.mark_errorband(
-        extent="ci",
-        opacity=0.1,
-        interpolate="step",
-    ).encode(
-        y=alt.Y("0.025:Q", axis=y_axis, scale=alt.Scale(type=scale)),
-        y2="0.975:Q",
-        fill=alt.value("steelblue"),
-    )
-    band_80 = base.mark_errorband(
-        extent="ci",
-        opacity=0.2,
-        interpolate="step",
-    ).encode(
-        y=alt.Y("0.10:Q", axis=y_axis, scale=alt.Scale(type=scale)),
-        y2="0.90:Q",
-        fill=alt.value("steelblue"),
-    )
-    band_50 = base.mark_errorband(
-        extent="iqr",
-        opacity=0.3,
-        interpolate="step",
-    ).encode(
-        y=alt.Y("0.25:Q", axis=y_axis, scale=alt.Scale(type=scale)),
-        y2="0.75:Q",
-        fill=alt.value("steelblue"),
+
+    def quantile_rename_map(cols: list[str]) -> dict[str, str]:
+        """
+        Creates a map for renaming quantile columns
+        formatted as strings.
+
+        Parameters
+        ----------
+        cols : list[str]
+            Column names from a (pivoted) forecast table.
+
+        Returns
+        -------
+        dict[str, str]
+            Mapping from original names to normalized
+            names with a "q" prefix and the leading
+            0 stripped. Columns not starting with 0 are
+            omitted.
+        """
+        return {
+            c: f"q{c[2:]}"
+            for c in cols
+            if isinstance(c, str) and c.startswith("0.")
+        }
+
+    rename_map = quantile_rename_map(df_wide.columns)
+    df_wide = df_wide.rename(rename_map)
+
+    x_enc = alt.X("target_end_date:T", title="Date", axis=alt.Axis(grid=grid))
+    y_enc = alt.Y(
+        "q5:Q",
+        axis=alt.Axis(grid=grid),
+        scale=alt.Scale(type=scale),
     )
+    base = alt.Chart(df_wide, width=PLOT_WIDTH).encode(x=x_enc, y=y_enc)
+
+    def band(width: float, opacity: float) -> alt.Chart:
+        """
+        Builds an errorband layer for a quantile.
+
+        Parameters
+        ----------
+        width : float
+            Interval width in (0, 1], e.g., 0.95 for a 95%
+            band (q025-q975), 0.80 for 80% (q1-q9), etc...
+        opacity : float
+            Fill opacity for the band in the range
+            [0.0, 1.0].
+
+        Returns
+        -------
+        alt.Chart
+            An Altair layer with the filled band from
+            ``low`` to ``high``, with step interpolation.
+        """
+        if not (0 < width <= 1):
+            raise ValueError("width must be in (0, 1].")
+
+        low = (1.0 - width) / 2.0
+        high = 1.0 - low
+
+        def norm_key(x: float) -> str:
```

Comment:
What is the intention of this function?

Reply:
Ah, this was an unforeseen complication on my part.

Let's restore the previous version of `band` which won't have these edge cases. Sorry about that.

https://github.com/CDCgov/hubverse-annotator/pull/136/files/5b64ba4b0720a0171808cfdf3650b106d6c20905#diff-5b5e6834e0a4c1d9059632c6de4d5ce7d205097342bd193c26fee21b6bddc6cfR297-R327